app/controllers/api/automate_workspaces_controller.rb

```diff
@@ -4,5 +4,32 @@ def edit_resource(type, id, data = {})
       obj = resource_search(id, type, collection_class(type))
       obj.merge_output!(data)
     end
+
+    def decrypt_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
```

Comment:
s/starting/decrypting/

Reply:
Also, is it possible for this error to raise? This needs a test proving why it is needed (as does the one below)